nixos/modules/services/security/vaultwarden/default.nix

```diff
@@ -221,6 +250,28 @@ in {
         User = mkDefault user;
         Group = mkDefault group;
         ExecStart = "${pkgs.bash}/bin/bash ${./backup.sh}";
+
+        ReadOnlyDirectories = mkIf (!hasPrefix "/var/lib/bitwarden_rs" cfg.backupDir) [ "/var/lib/bitwarden_rs" ];
```

Comment:
Why do we only do this if the prefix is bitwarden_rs? I have changed the stateDirectory to be vaultwarden. Would this then break?

Reply:
I don't remember my reason for doing this very clearly, but there's a small `!` in there. I think the idea was that the backup script should not be able to modify any of the data it's copying. But keeping backups in `/var/lib/bitwarden_rs/backup` and marking `/var/lib/bitwarden_rs` as `ReadOnly` caused issues, hence the `hasPrefix`.

Reply:
I think it is alright if it can read and write all the data. It is a big improvement from the status quo regardless.